src/base/system.h

```diff
@@ -2230,6 +2231,14 @@ unsigned long str_toulong_base(const char *str, int base);
 int64_t str_toint64_base(const char *str, int base = 10);
 float str_tofloat(const char *str);
 
+template<typename T>
+void str_format_integer(char *buffer, size_t buffer_size, T value, int base = 10)
+{
+	buffer[0] = '\0'; // Fix false positive clang-analyzer-core.UndefinedBinaryOperatorResult when using result
+	auto result = std::to_chars(buffer, buffer + buffer_size - 1, value, base);
+	result.ptr[0] = '\0';
+}
```

Comment:
As far as I see, in all cases we write to [0] of a buffer. It means that we can safely use a templated function and we don't even need the unsafe one (with manually specified buffer size).

```suggestion
template<typename T, size_t N>
void str_format_integer(char (&buffer)[N], T value, int base = 10)
{
	buffer[0] = '\0'; // Fix false positive clang-analyzer-core.UndefinedBinaryOperatorResult when using result
	auto result = std::to_chars(buffer, buffer + N - 1, value, base);
	result.ptr[0] = '\0';
}
```

Reply:
I added the templated function but kept the unsafe one, to avoid adding `#include <charconv>` in `system.h`.

Reply:
👍 after reading @heinrich5991 comment I thought we'd better to do that trade of (add a template but keep the function with `size_t buffer_size` to move the header to `.cpp`) and I also thought you'll see it without my comments 😄.